client/test/unit/specs/components/JobList.spec.js

```diff
@@ -186,6 +191,139 @@ describe('Unit | Component | JobList.vue', () => {
         expect(jobTitles[1].textContent).to.equal('Tech Lead mission 1');
       }));
     });
+
+    describe('after jobs are loaded, a country filter is applied', () => {
+      beforeEach(() => {
+        // given
+        jobs = [
+          {
+            id: 1,
+            activity: {
+              title: 'Tech Lead mission 1',
+            },
+            project: {
+              status: 'proposal_in_progress',
+              customer: {
+                sector: {
+                  name: 'FR - La Poste',
+                },
+              },
+            },
+          },
+          {
+            id: 2,
+            activity: {
+              title: 'Tech Lead mission 2',
+            },
+            project: {
+              status: 'mission_signed',
+              customer: {
+                sector: {
+                  name: 'Australia',
+                },
+              },
+            },
+          },
+        ];
+        expectedJobs = [
+          {
+            id: 2,
+            activity: {
+              title: 'Tech Lead mission 2',
+            },
+            project: {
+              name: 'Spark/Hive Archi for eCRM',
+              status: 'mission_signed',
+              customer: {
+                sector: {
+                  name: 'Australia',
+                },
+              },
+              business_contact: {
+                nickname: 'ABC',
+              },
+            },
+          },
+          {
+            id: 1,
+            activity: {
+              title: 'Tech Lead mission 1',
+            },
+            project: {
+              name: 'Système d\'excellence',
+              status: 'proposal_in_progress',
+              customer: {
+                sector: {
+                  name: 'FR - La Poste',
+                },
+              },
+              business_contact: {
+                nickname: 'ABC',
+              },
+            },
+          },
+        ];
+
+        sinon.stub(projectStatus, 'sort').returns(expectedJobs);
+        sinon.stub(jobsApi, 'fetchAll').resolves(jobs);
+
+        // when
+        component = new Constructor().$mount();
+      });
+
+      afterEach(() => {
+        projectStatus.sort.restore();
+        jobsApi.fetchAll.restore();
+      });
+
+      it('should render as many jobs as received from the API', () => Vue.nextTick().then(() => {
+        expect(getJobCardsCount(component)).to.equal(2);
+      }));
+
+      it('When selecting jobs in France and overseas, should display the two listed jobs', () => Vue.nextTick().then(() => {
```

Comment:
ajouter un describe par "when"
et conserver un it par "should"

Reply:
+ ne pas oublier de mettre des return devant les Vue.nextTick().then(() => {
sinon l'assertion n'est pas prise en compte lorsque les tests sont passés